homeassistant/components/sensor/linky.py

```diff
@@ -0,0 +1,84 @@
+"""
+Support for Linky.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/sensor.linky/
+"""
+import logging
+import json
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pylinky==0.1.5']
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'linky'
+
+SCAN_INTERVAL = timedelta(minutes=10)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Configure the platform and add the Linky sensor."""
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    from pylinky.client import LinkyClient
+    client = LinkyClient(username, password)
+
+    devices = [LinkySensor('Linky', client)]
+    add_devices(devices, True)
+
+
+class LinkySensor(Entity):
+    """Representation of a sensor entity for Linky."""
+
+    def __init__(self, name, client):
+        """Initialize the sensor."""
+        self._name = name
+        self._client = client
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return 'kWh'
+
+    @Throttle(SCAN_INTERVAL)
+    def update(self):
+        """Fetch new state data for the sensor."""
+        try:
+            self._client.fetch_data()
+        except BaseException as exp:
+            _LOGGER.error(exp)
+        finally:
+            self._client.close_session()
+
+        _LOGGER.debug(json.dumps(self._client.get_data(), indent=2))
+
+        if self._client.get_data():
```

Comment:
Handle this in the try-except block. Return as early as possible.

Reply:
`get_data()` is a getter, not an API call (`fetch_data()` is). Are you sure to want to put it into the try/catch block?

Reply:
If `fetch_data()` fails then it doesn't make sense to go on with `update()`.